src/status_im/contexts/wallet/wallet_connect/core.cljs

```diff
@@ -50,7 +50,11 @@
 
 (defn get-session-dapp-metadata
   [proposal]
-  (get-in proposal [:params :proposer :metadata]))
+  (let [metadata (get-in proposal [:params :proposer :metadata])
+        origin   (get-in proposal [:verifyContext :verified :origin])]
```

Comment:
Can get the url right here:

```clj
(get-in proposal [:verifyContext :verified :origin :url])
```

and later instead of `{:url origin}`, just `origin`

Reply:
the origin is the url and just to create the same structure like :metadata i put it like this {:url origing}, we use this func like this
`[{:keys [name url]} (wallet-connect-core/get-session-dapp-metadata proposal)]`